sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/constraints.scala

```diff
@@ -112,6 +117,27 @@ case class CheckConstraint(
   with TableConstraint {
 // scalastyle:on line.size.limit
 
+  def toV2Constraint(isCreateTable: Boolean): Constraint = {
```

Comment:
I wonder if the input param should be related to the validation status, rather than to whether it is create or alter. For instance, we can make validation optional in ALTER.

Reply:
ok, how about let's make all the validate status as `UNVALIDATED` in this PR? Once we support enforcing check constraint, we can have more discussions on this one